### PR TITLE
add shib auth support for redirect url

### PR DESCRIPTION
**Pull request by mrow4a**
## Description
Without this change, server-side Apache Auth (e.g. Shibboleth) resulted in redirect to default owncloud page, instead of last visited page. We now correctly handle redirect_url. 

## Related Issue
- Fixes https://github.com/owncloud/enterprise/issues/5450
- Related https://github.com/owncloud/core/pull/40161

## How Has This Been Tested?
- manually in internal oc2.lab test instance

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] Base code changes
- [x] Make sure to handle corner cases in login
- [x] Unit tests added
- [x] Acceptance tests added
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiProvisioning-v2-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/37175/43

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUILogin-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/37175/125

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>cliTrashbin-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/37175/110

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingPublic2-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/37175/142

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiShareManagementBasicToRoot-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/37185/53

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiShareOperationsToRoot1-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/37185/55

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiShareManagementToRoot-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/37185/51

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiShareManagementToShares-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/37185/52

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]

File: core/templates/apacheauthredirect.php [hunk 1]
```diff
@@ -5,6 +5,8 @@
 ]);
 ?>
 
-<span class="msg">
-	<?php p($l->t('The application was authorized successfully. You can now close this window.'));?>
-</span>
+<?php print_unescaped('<input type="hidden" id="redirect_url" name="redirect_url" value="' . \OCP\Util::sanitizeHTML($_['redirect_url']) . '">'); ?>
+
+<p class="info">
+	<?php p($l->t('The application was authorized successfully. You should now get redirected to the requested page, otherwise you can close this window.'));?>
```
[the remark this answers is not in the document]
Comment (mrow4a) [reply core/templates/apacheauthredirect.php]: @DeepDiver1975 done

**Review by DeepDiver1975 — APPROVED** [review]
